server.js

```diff
@@ -1,29 +1,98 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
+import express from 'express';
+import bodyParser, { json } from 'body-parser';
+import cors from 'cors';
+import mongoose from 'mongoose';
 
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
+const mongoUrl = process.env.MONGO_URL || 'mongodb://localhost/happyThoughts';
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
+mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+
+//Mongoose model
+const Thought = new mongoose.model('Thought', {
+	message: {
+		type: String,
+		required: true,
+		minlength: 5,
+		maxlength: 140,
+	},
+	hearts: {
+		type: Number,
+		default: 0,
+	},
+	createdAt: {
+		type: Date,
+		default: Date.now,
+	},
+});
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+	res.send('Welcome to Happy happy place');
+});
+
+app.get('/thoughts', async (req, res) => {
+	try {
+		const thoughts = await Thought.find()
+			.sort({ createdAt: 'desc' })
+			.limit(20)
+			.exec();
+
+		res.json(thoughts);
+	} catch (err) {
+		res.status(400).json({
+			success: false,
+			message: 'Could not get thoughts',
+			errors: err.errors,
+		});
+	}
+});
+
+//Post a new thought
+
+app.post('/thoughts', async (req, res) => {
+	const { message } = req.body;
+	const newThought = await new Thought({ message });
+	try {
+		const savedThought = await newThought.save();
+		res.status(201).json(savedThought);
```

Comment:
Nice job with the status messages, makes the code easy to read and understand! :)